51/vipinreyo/nba.py

```diff
@@ -0,0 +1,166 @@
+                        avg_min=row['Minutes.per.Game'],
+                        avg_points=row['Points.per.Game'])
+        yield player
+
+
+# CODE HERE (tests under __main__):
+
+def import_to_db(players=None):
+    """Create database table in sqlite3 and import the players data
+
+       required table SQL:
+       CREATE TABLE players (name, year, first_year, team, college,
+                             active, games, avg_min, avg_points)
+    """
+    if players is None:
+        players = list(load_data())
+
+    # Create the table db statement
+    players_table = """CREATE TABLE players (
+                            name text,
+                            year integer,
+                            first_year integer,
+                            team text,
+                            college text,
+                            active integer,
+                            games integer,
+                            avg_min real,
+                            avg_points real
+                            ); """
+
+    cur.execute(players_table)
+
+    # Insert players to the table
+    with conn:
+        cur.executemany("insert into players (name, year, first_year, team, college, active, games,"
+                            " avg_min, avg_points) values (?, ?, ?, ?, ?, ?, ?, ?, ?)", players)
+
+
+def player_with_max_points_per_game():
+    """The player with highest average points per game (don't forget to CAST to
+       numeric in your SQL query)"""
+    sql = ''' select name from players group by games order by avg_points desc limit 1;'''
+    cur.execute(sql)
+    result = cur.fetchall()
+    return result[0][0]
+
+
+def number_of_players_from_duke():
+    """Return the number of players with college == Duke University"""
+    sql = '''select count(*) from players where college == 'Duke University';'''
+    cur.execute(sql)
+    result = cur.fetchall()
+    return result[0][0]
+
+
+def percentage_of_players_first_year():
+    """Return 2 digit percentage of players whose first year it is
+       (first_year column)"""
+    sql1 = '''select count(*) from players where first_year == 1'''
+    sql2 = '''select count(*) from players'''
+    cur.execute(sql1)
+    count_players_first_year = cur.fetchall()[0][0]
+    cur.execute(sql2)
+    count_players = cur.fetchall()[0][0]
+    return (count_players_first_year/count_players) * 100
+
+
+def avg_years_active_players_stanford():
+    """Return the average years that players from "Stanford University
+       are active ("active" column)"""
+    sql = '''select avg(active) from players where college == 'Stanford University';'''
+    cur.execute(sql)
+    result = cur.fetchall()
+    return result[0][0]
+
+
+def year_with_most_drafts():
+    """Return the year with the most drafts, in SQL you can use GROUP BY"""
+    sql = '''select year, count(*) as cnt from players group by year order by cnt desc limit 1;'''
+    cur.execute(sql)
+    result = cur.fetchall()
```

Comment:
cool, I did fetchall too til lI saw Martin use fetchone, which I might update the solution too ...